src/xma/src/xmaapi/xma_utils.cpp

```diff
@@ -550,46 +543,51 @@ int32_t check_all_execbo(XmaSession s_handle) {
             for (i = 0; i < num_execbo; i++) {
                 auto& ebo = priv1->kernel_execbos[i];
                 if (ebo.in_use) {
-                    ert_start_kernel_cmd *cu_cmd = 
-                        (ert_start_kernel_cmd*)ebo.data;
-                    if (cu_cmd->state == ERT_CMD_STATE_COMPLETED) {
+                    ert_start_kernel_cmd * cu_cmd_pkt =
+                        reinterpret_cast<ert_start_kernel_cmd*>(ebo.xrt_run.get_ert_packet());// only for PS Kernels, will be removed after PS kernel migration to xrt::kernel   
+                    if (!cu_cmd_pkt) {
+                        xma_logmsg(XMA_ERROR_LOG, XMAUTILS_MOD, "check_all_execbo : Invalid ert_start_kernel_cmd\n");
+                        return XMA_ERROR;
+                    }
+                    auto cu_state = ebo.xrt_run.state();
+                    if (cu_state == ERT_CMD_STATE_COMPLETED) {
```

Comment:
we may need to stop understanding this ERT* enums. This can be in next PR, Please take a note.

Reply:
sure vamshi.